lib/dogma/rules/hard_tabs.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Dogma.Rules.HardTabs do
+  @moduledoc """
+  Requires that all indentation is done using spaces rather than hard tabs.
+
+  So the following would be invalid:
+
+  def something do
+  \t:body
+  end
+  """
+  @behaviour Dogma.Rule
+
+  @indenting_tab_pattern ~r/^\t+.+/
```

Comment:
I considered just looking for tabs in general but I thought there might be some legitimate uses cases. Looking for them at the start will spot most of the invalid cases and hopefully without any of the false positives.